base/number.jl

```diff
@@ -63,4 +63,6 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 one(x::Number)  = oftype(x,1)
 one{T<:Number}(::Type{T}) = convert(T,1)
 
+_default_type(::Type{Number}) = Int
```

Comment:
(I would make the same recommendation for your other PRs. e.g. separate the PR to add tuple support to broadcast from the PR to generalize the concept of scalar.)